arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -923,6 +924,10 @@ Result RocksDBCollection::update(arangodb::transaction::Methods* trx,
     return res;
   }
 
+  bool useRev = _logicalCollection.version() >= LogicalCollection::Version::v36;
+  LocalDocumentId const newDocumentId =
+      useRev ? LocalDocumentId::create(revisionId) : LocalDocumentId::create();
+
```

Comment:
I think it is effectively not an issue to unconditionally use the revision (except in the smart edge case)

LocalDocumentId and the revisions are generated from the same server-global HLC counter

Reply:
i.e. TRI_HybridLogicalClock

Reply:
in any case, perhaps we should put this in a helper method ?

Reply:
Using a helper method seems reasonable. I'll make the change.